app/src/main/java/org/simple/clinic/appconfig/StatesFetcher.kt

```diff
@@ -0,0 +1,35 @@
+package org.simple.clinic.appconfig
+
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.simple.clinic.appconfig.api.StatesApi
+import retrofit2.Retrofit
+import java.net.ConnectException
+import javax.inject.Inject
+
+class StatesFetcher @Inject constructor(
+    private val retrofitBuilder: Retrofit.Builder
```

Comment:
> Since it's possible for the underlying `StatesApi` call to throw connection exception we are marking it for the `fetchStates` call, so that mockito can throw the exception properly.

@msasikanth can you please elaborate on this more? In what cases we have a connection exception? Are we handling this in this case because we are going to be choosing a different server end point which can be rejected? Why are we not handling this when we are initially connected to different servers based on the country?

Reply:
>Are we handling this in this case because we are going to be choosing a different server endpoint that can be rejected?

No, basically the API call can throw a `ConnectionException`, it might be because of some connection issues to the server. But since this method is not marked as a can throw an error, Mockito was not able to resolve the connection exception test. So, I have added that to the public method we are exposing.

There might be a better way of doing this, but at this moment this was the best I can think of. @vinaysshenoy can you take a look at it and see if there was a better way of handling it? 🤔  (There won't be any changes to the way it functions in prod, this is just for testing)